### PR TITLE
Pre-main job to build a kyma-cli based on cli:main and reconciler:PR

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Pre-main job to build a kyma-cli based on cli:main and reconciler:PR
- [Test run](https://status.build.kyma-project.io/view/gs/kyma-prow-logs/pr-logs/pull/kyma-incubator_reconciler/899/mfaizanse_test_of_prowjob_pre-main-reconciler-publish-pr-cli/1501851604922404864)

**Related issue(s)**

- https://github.com/kyma-incubator/reconciler/issues/802

### CODE REVIEW

File: prow/scripts/reconciler-publish-pr-cli.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+RECONCILER_PROJECT_DIR="/home/prow/go/src/github.com/kyma-incubator/reconciler"
+
+# shellcheck source=prow/scripts/lib/log.sh
+source "${SCRIPT_DIR}/lib/log.sh"
+# shellcheck source=prow/scripts/lib/utils.sh
+source "${SCRIPT_DIR}/lib/utils.sh"
+# shellcheck source=prow/scripts/lib/gcp.sh
+source "$SCRIPT_DIR/lib/gcp.sh"
+
+cd "${KYMA_CLI_PROJECT_DIR}"
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply prow/scripts/reconciler-publish-pr-cli.sh:17]: IMO, it's better to do it in the bash script for readability, we would be sure we are in the right directory.

File: prow/scripts/reconciler-publish-pr-cli.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+log::banner "Replacing reconciler dependency used by the Kyma CLI to local PR source"
+go mod edit -replace "github.com/kyma-incubator/reconciler=${RECONCILER_PROJECT_DIR}"
+
+log::info "Contents of kyma-project/cli/go.mod"
+cat go.mod
+
+log::banner "Building Kyma CLI with recocniler bump from PR source"
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply prow/scripts/reconciler-publish-pr-cli.sh]: updated

File: prow/scripts/reconciler-publish-pr-cli.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+make build-linux
+
+log::info "Renaming Kyma CLI to include PR number"
+mv "./bin/kyma-linux" "./bin/kyma-linux-pr-${PULL_NUMBER}"
+ls "./bin/"
+
+log::info "Committing reconciler bump"
+git_status=$(git status --porcelain)
+if [[ "${git_status}" != "" ]]; then
+  git commit -am 'replace reconciler version'
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply prow/scripts/reconciler-publish-pr-cli.sh]: I have removed it.

**Review by k15r — APPROVED** [review]

**Review by ammarlakis — APPROVED** [review]

**Pull request merged**